opal/mca/memory/linux/hooks.c

```diff
@@ -734,7 +735,10 @@ static check_result_t check(const char *name)
     }
 }
 
-/* OMPI's init function */
+
+/* This function is called from MPI_Init in case system has Memory Allocation Hooks
+ * (see ompi/runtime/ompi_mpi_init.c)
+ */
```

Comment:
Ok.  Should probably update the comment to be a bit more generic.  And per my other comments, we might want to make this functionality opt-in vs. global.